src/OpenTelemetry.Exporter.Jaeger/JaegerExporter.cs

```diff
@@ -0,0 +1,128 @@
+﻿// <copyright file="JaegerExporter.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Exporter.Jaeger
+{
+    using System;
+    using OpenTelemetry.Exporter.Jaeger.Implimentation;
+    using OpenTelemetry.Trace.Export;
+
+    public class JaegerExporter : IDisposable
+    {
+        public const string DefaultAgentUdpHost = "localhost";
+        public const int DefaultAgentUdpCompactPort = 6831;
+        public const int DefaultMaxPacketSize = 65000;
+
+        private const string ExporterName = "JaegerTraceExporter";
+
+        private readonly object @lock = new object();
+        private readonly JaegerExporterOptions options;
+        private readonly IExportComponent exportComponent;
+
+        private bool isInitialized = false;
+        private JaegerTraceExporterHandler handler;
+        private bool disposedValue = false; // To detect redundant dispose calls
+
+        public JaegerExporter(JaegerExporterOptions options, IExportComponent exportComponent)
+        {
+            this.ValidateOptions(options);
+            this.InitializeOptions(options);
+
+            this.options = options;
+            this.exportComponent = exportComponent;
+        }
+
+        public void Start()
+        {
+            lock (this.@lock)
+            {
+                if (this.isInitialized)
+                {
+                    return;
+                }
+
+                if (this.exportComponent != null)
+                {
+                    this.handler = new JaegerTraceExporterHandler(this.options);
+                    this.exportComponent.SpanExporter.RegisterHandler(ExporterName, this.handler);
+                }
+            }
+        }
+
+        public void Stop()
+        {
+            if (!this.isInitialized)
```

Comment:
I might be totally wrong here so apologies in advance, but: Wouldn't it be possible for this value to be read from a CPU cache and be outdated? Since it's outside a memory barrier.
Would it need to be `volatile` in this case? Only for the same of a double check lock which isn't being used here since the value isn't checked again inside the `lock`.

Reply:
yes, should be volatile. thanks.

Reply:
Cool. So it'll need a double check lock, right? I mean the volatile read will ensure no CPU cache involved but inside the `lock` another look at `if (!this.isInitialized)` would need to be executed.